app/views/products/_form.html.erb

```diff
@@ -0,0 +1,58 @@
+<div class="custom-border">
+  <%= form_with model: @product do |form| %>
+    <div class="mb-3">
+      <%= form.label :title, class: "form-label" %>
+      <%= form.text_field :title, class: "form-control" %>
+      <% product.errors.full_messages_for(:title).each do |message| %>
```

Comment:
use a method or use this logic in a helper, product.errors.full_messages_for, since you're repeating it again and again.